src/intrinsics/ecma262/Object.js

```diff
@@ -157,12 +157,10 @@ export default function(realm: Realm): NativeFunctionValue {
 
       // If we generated an Object.assign() to deal with partials, by this
       // point it is not safe to interact with those objects in Prepack land.
+      Leak.leakValue(to);
```

Comment:
I am still not sure that the target needs to leak. Please explain why in a comment.